repo/object/object_manager.go

```diff
@@ -44,17 +44,29 @@ type contentManager interface {
 type Manager struct {
 	Format format.ObjectFormat
 
-	contentMgr  contentManager
-	newSplitter splitter.Factory
-	writerPool  sync.Pool
+	contentMgr         contentManager
+	newDefaultSplitter splitter.Factory
+	writerPool         sync.Pool
 }
 
 // NewWriter creates an ObjectWriter for writing to the repository.
 func (om *Manager) NewWriter(ctx context.Context, opt WriterOptions) Writer {
 	w, _ := om.writerPool.Get().(*objectWriter)
 	w.ctx = ctx
 	w.om = om
-	w.splitter = om.newSplitter()
+
+	var splitFactory splitter.Factory
+
+	if opt.Splitter != "" {
+		splitFactory = splitter.GetFactory(opt.Splitter)
```

Comment:
nit: if the specified splitter is not found, then this silently falls back to the default. This may be confusing since there's no error or warning. Is this logged anywhere?